js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -163,24 +170,27 @@ export function useExecutionSessionStorage(
 ): StorageHook {
   const {basePath} = React.useContext(AppContext);
 
-  const oldNamespace = `${repoAddress.name}.${pipelineOrJobName}`;
-  const oldData = getStorageDataForNamespace(oldNamespace);
-
-  const namespace = `${basePath}-${repoAddress.location}-${repoAddress.name}-${pipelineOrJobName}`;
-  const [version, setVersion] = React.useState<number>(0);
+  const namespace = makeNamespace(basePath, repoAddress, pipelineOrJobName);
+  const [_, setVersion] = React.useState<number>(0);
 
-  const onSave = (newData: IStorageData) => {
-    writeStorageDataForNamespace(namespace, newData);
-    setVersion(version + 1); // trigger a React render
-  };
+  const onSave = React.useCallback(
+    (newData: IStorageData) => {
+      writeStorageDataForNamespace(namespace, newData);
+      setVersion((current) => current + 1); // trigger a React render
+    },
+    [namespace],
+  );
 
-  // TODO: Remove this migration logic in a few patches when we know the old namespace is likely no longer being used
-  const oldDataMigrated = React.useRef(false);
-  if (oldData && !oldDataMigrated.current) {
-    onSave(oldData);
-    window.localStorage.removeItem(getKey(oldNamespace));
-    oldDataMigrated.current = true;
-  }
+  // Legacy namespace. Use this to migrate from old storage namespace to new one.
+  const oldNamespace = makeOldNamespace(repoAddress, pipelineOrJobName);
+  React.useEffect(() => {
+    const oldKey = getKey(oldNamespace);
```

Comment:
This is unrelated to this change, but I think having both "data->namespace" and "namespace->key" transforms makes this a bit harder to follow? Maybe the dagit.v2 prefix could be added at the same time and we could boil these down to:

- `keyForSessionStorage(basePath, repoAddress, pipelineOrJobName)`
- `oldKeyForSessionStorage(repoAddress, pipelineOrJobName)`